gemrb/core/Audio/MusicLoop.cpp

```diff
@@ -0,0 +1,142 @@
+/* GemRB - Infinity Engine Emulator
+ * Copyright (C) 2025 The GemRB Project
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+#include <chrono>
+
+#include "Interface.h"
+#include "MusicLoop.h"
+#include "MusicMgr.h"
+
+namespace GemRB {
+
+MusicLoop::MusicLoop()
+{
+	auto audio = core->GetAudioDrv();
+	streamHandle =
+		audio->CreateStreamable(
+			core->GetAudioSettings().ConfigPresetMusic()
+		);
+
+	needToPoll = audio->GetStreamMode() == AudioBackend::StreamMode::POLLING;
+	loopThread = std::thread(&MusicLoop::Loop, this);
+}
+
+MusicLoop::~MusicLoop()
+{
+	streamHandle->Stop();
+	loop = false;
+	loopThread.join();
+}
+
+void MusicLoop::Load(ResourceHolder<SoundMgr> music)
+{
+	std::lock_guard<std::recursive_mutex> l(mutex);
+
+	currentFormat.bits = 16;
+	currentFormat.channels = 2; // get_channels may report garbage
+	currentFormat.sampleRate = music->get_samplerate();
+
+	loadBuffer.resize(currentFormat.GetNumBytesForMs(125) / 2);
```

Comment:
Is there any particular significance to 125? I suggest you add a comment why we use 1/8s, so it will be easier to reason about if we ever need to revisit this.

Reply:
By a recent talk by Valve 😄 It was mentioned how they manage sound, such that 125ms was seen as a balance between a decent memory buffer amount that is usually long enough to cover a high percentile of I/O latency spikes on low machines without interruption. And since I was out of competing models ...

Unless an Raspberry Pi starts to struggle here, I'd commit it like "educated guess".

Reply:
Would be fine as a comment, it being somewhat of an industry standard is very good justification.